.github/workflows/ci.yml

```diff
@@ -8,19 +8,48 @@ on:
       - 'v*.*'
   pull_request:
 
+# Cancels all previous workflow runs for pull requests that have not completed.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+  cancel-in-progress: true
+
 jobs:
-  test:
+  checks:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
 
+      # Initialize Cache Composer
+      - name: Cache Composer dependencies
+        uses: actions/cache@v3
+        with:
+          path: .cache
+          key: ${{ runner.os }}-${{ hashFiles('**/composer.lock') }}
+
+      # Initialize Composer
+      # Will throw error to upgrade to Composer V2.
+      # We stick to V1 because of PHP 7.3 usage. https://github.com/marketplace/actions/composer-update-action
+      - uses: php-actions/composer@v6
+        env:
+          COMPOSER: 'composer.json'
+        with:
+          php_version: '7.2'
+          version: 1
+
+      # Initialize Node.js
       - uses: actions/setup-node@v3
         with:
           node-version: '20'
           check-latest: true
           cache: npm
           cache-dependency-path: package-lock.json
 
-      - run: npm ci
+      # Make sure .cache folder exists
+      - name: 'Create folder .cache s'
+        run: |
+          mkdir -p .cache && cd .cache && touch .gitkeep
```

Comment:
Why is this needed? Isn't the `.cache` dir already created by Composer above? PHPCS might also create it.

Reply:
Is needed in case the folder do not exist on repository